pkg/provider/consulcatalog/connect_tls.go

```diff
@@ -52,15 +53,20 @@ func (c *connectCert) equals(other *connectCert) bool {
 }
 
 func (c *connectCert) serversTransport(item itemData) *dynamic.ServersTransport {
+	itemName := item.Name
+	if item.ExtraConf.ConsulNameSuffix != "" {
+		itemName = strings.TrimSuffix(item.Name, "-"+item.ExtraConf.ConsulNameSuffix)
```

Comment:
This smells like a hack :D I think it would make more sense to add a `ConsulServiceName` to `itemData` so you always have access to the original unmodified service name instead of trimming here.

Reply:
@apollo13 I updated the PR to store the original ConsulServiceName on itemData and use that where the unmodified consul service name is needed